test/util.test.ts

```diff
@@ -68,3 +72,391 @@ describe('ensureArgObject', () => {
+  })
+})
+
+describe('isTruthy', () => {
+  it('should return true for truthy values', () => {
+    expect(isTruthy('true')).to.be.true
+    expect(isTruthy('1')).to.be.true
+    expect(isTruthy('yes')).to.be.true
+    expect(isTruthy('y')).to.be.true
+  })
+
+  it('should return false for falsy values', () => {
+    expect(isTruthy('false')).to.be.false
+    expect(isTruthy('0')).to.be.false
+    expect(isTruthy('no')).to.be.false
+    expect(isTruthy('n')).to.be.false
+  })
+})
+
+describe('getHomeDir', () => {
+  it('should return the home directory', () => {
+    expect(getHomeDir()).to.equal(homedir())
+  })
+})
+
+describe('readJson', () => {
+  it('should return parsed JSON', async () => {
+    const json = await readJson<{name: string}>('package.json')
+    expect(json.name).to.equal('@oclif/core')
+  })
+
+  it('should throw an error if the file does not exist', async () => {
+    try {
+      await readJson('does-not-exist.json')
+      throw new Error('Expected an error to be thrown')
+    } catch (error) {
+      const err = error as Error
+      expect(err.message).to.include('ENOENT: no such file or directory')
+    }
+  })
+})
+
+describe('castArray', () => {
+  it('should cast a value to an array', () => {
+    expect(castArray('foo')).to.deep.equal(['foo'])
+  })
+
+  it('should return an array if the value is an array', () => {
+    expect(castArray(['foo'])).to.deep.equal(['foo'])
+  })
+
+  it('should return an empty array if the value is undefined', () => {
+    expect(castArray()).to.deep.equal([])
+  })
+})
+
+describe('defaultToCached', () => {
+  it('should do nothing if noCacheDefault is true', async () => {
+    const flag = {noCacheDefault: true}
+    const result = await defaultToCached(flag as any, true)
+    expect(result).to.be.undefined
+  })
+
+  it('should do nothing if respectNoCacheDefault is true', async () => {
+    const result = await defaultToCached({} as any, true)
+    expect(result).to.be.undefined
+  })
+
+  it('should return the result of defaultHelp if it exists', async () => {
+    const flag = {defaultHelp: async () => 'foo'}
+    const result = await defaultToCached(flag as any, false)
+    expect(result).to.equal('foo')
+  })
+
+  it('should return undefined if defaultHelp throws', async () => {
+    const flag = {async defaultHelp() {
+      throw new Error('foo')
+    }}
+    const result = await defaultToCached(flag as any, false)
+    expect(result).to.be.undefined
+  })
+
+  it('should return the result of the default if it\'s a function', async () => {
+    const flag = {default: async () => 'foo'}
+    const result = await defaultToCached(flag as any, false)
+    expect(result).to.equal('foo')
+  })
+
+  it('should return the result of the default if it\'s a simple value', async () => {
+    const flag = {default: 'foo'}
+    const result = await defaultToCached(flag as any, false)
+    expect(result).to.equal('foo')
+  })
+})
+
+describe('toCached', () => {
```

Comment:
maybe this should be its own file to mirror that util/to-cached.ts is a single file.

Easier for someone messing with that util to know what test they need to be watching?